src/cdk-experimental/selection/selection-toggle.ts

```diff
@@ -100,5 +100,14 @@ export class CdkSelectionToggle<T> implements OnDestroy, OnInit {
     return this._selection.isSelected(this.value, this.index);
   }
 
+  // This has been extracted to a util because of TS 4 and VE.
+  // View Engine doesn't support property rename inheritance.
+  // TS 4.0 doesn't allow properties to override accessors or vice-versa.
+  protected static _setIndexInput<T>(
```

Comment:
Why make these functions `static` instead of just a regular class method?

Reply:
So they don’t show up in the IDE during auto completion. I change this, though @crisbeto suggested this approach.

Reply:
That makes sense, but we already add a bunch of "non-public" members to classes, so I don't want to start doing something different now. I'd rather take a more holistic approach and introduce tooling to remove these non-public APIs from the `d.ts` files we publish to npm

Reply:
I think we should investigate tsc `@internal` and `stripInternal` options in future.